contrib/epee/src/wipeable_string.cpp

```diff
@@ -109,9 +115,112 @@ void wipeable_string::push_back(char c)
   buffer.back() = c;
 }
 
-void wipeable_string::pop_back()
+void wipeable_string::operator+=(char c)
+{
+  push_back(c);
+}
+
+void wipeable_string::operator+=(const std::string &s)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - s.size(), "Appended data too large");
+  grow(size(), size() + s.size());
+  const char *ptr = s.c_str();
+  while (*ptr)
+    push_back(*ptr++);
+}
+
+void wipeable_string::operator+=(const epee::wipeable_string &s)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - s.size(), "Appended data too large");
+  grow(size(), size() + s.size());
+  const char *ptr = s.data();
+  size_t sz = s.size();
+  while (sz--)
+    push_back(*ptr++);
+}
+
+void wipeable_string::operator+=(const char *s)
+{
+  size_t slen = strlen(s);
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - slen, "Appended data too large");
+  grow(size(), size() + slen);
+  while (*s)
+    push_back(*s++);
+}
+
+void wipeable_string::append(const char *data, size_t len)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - len, "Appended data too large");
+  grow(size(), size() + len);
+  while (len--)
+    push_back(*data++);
+}
+
+void wipeable_string::trim()
+{
+  size_t prefix = 0;
+  while (prefix < size() && data()[prefix] == ' ')
+    ++prefix;
+  if (prefix > 0)
+    memmove(buffer.data(), buffer.data() + prefix, size() - prefix);
+
+  size_t suffix = 0;
+  while (suffix < size()-prefix && data()[size() - 1 - prefix - suffix] == ' ')
+    ++suffix;
+
+  resize(size() - prefix - suffix);
+}
+
+void wipeable_string::split(std::vector<wipeable_string> &fields)
+{
+  fields.clear();
+  size_t len = size();
+  const char *ptr = data();
+  bool space = true;
+  while (len--)
+  {
+    const char c = *ptr++;
+    if (c != ' ')
+    {
+      if (space)
+        fields.push_back({});
+      fields.back().push_back(c);
+    }
+    space = c == ' ';
+  }
+}
+
+bool wipeable_string::parse_hexstr(wipeable_string &res)
+{
+  res.clear();
+  if (size() & 1)
+    return false;
+  const size_t len = size();
+  const char *d = data();
+  res.grow(0, len / 2);
+  static constexpr const char hex[] = "0123456789abcdef";
```

Comment:
You're concerned the compiler would sneak in an EBCDIC string here ?
Can you explain what that u8 prefix is again ? You might have before, but I'd never seen that before you wanted these in, and there are lots of places in the source where strings are used without this.

Reply:
I've found an explanation of what u8 is for.

Reply:
The original implementation you wrote would not work if the locale used by `cin` was incompatible with the system C locale (the string constant `hex` would be in a different encoding than the string from `cin`). The `epee` implementation that is in use now uses `strtoul`, which should work even if the `cin` locale is not ASCII compatible. Given that this function lacks a specific context, I thought it made sense to make this implementation unambiguously for ASCII/UTF8 inputs only - getting it to work with multiple locales required more work anyway.

FWIW, the `epee` conversion has an issue because its used in JSON situations, which is always going to be UTF8 from rapidjson now. No one has reported this issue - presumably the bulk of computers are using UTF8 locally now.

Reply:
OK, I don't understand what you're saying, so I'll just change it since you seem to be sure.